PyInstaller/hooks/hook-keyring.py

```diff
@@ -0,0 +1,15 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2013-2018, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License with exception
+# for distributing bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from PyInstaller.utils.hooks import collect_submodules, copy_metadata
+
+hiddenimports = collect_submodules('keyring.backends')
+
+# keyring uses entrypoints to read keyring.backends from metadata file entry_points.txt.
+datas = copy_metadata('keyring')
```

Comment:
Surely for a backend to declare itself, its metadata needs to be collected too?

Reply:
The backends are "registered" by having their entry points listed in the (main package's) `entry_points.txt`, i.e.:
```
[keyring.backends]
KWallet = keyring.backends.kwallet
SecretService = keyring.backends.SecretService
Windows = keyring.backends.Windows
chainer = keyring.backends.chainer
macOS = keyring.backends.OS_X
```
That's why `keyring`'s metadata needs to be collected.

This PR is essentially the same as #5245 (I didn't notice it before), except that it also "upgrades" the hook from `keyring.backends` to `keyring` (which does not seem to be strictly necessary, though).

Reply:
Ahh, I always assumed entry points were organised the other way around. 

> it also "upgrades" the hook from keyring.backends to keyring

Unless it's possible to write code that uses `keyring` but doesn't use `keyring.backends` (even indirectly) then this has no effect really.